api/src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecution.java

```diff
@@ -211,4 +214,10 @@ public boolean blocksRestart() {
      */
     public abstract @Nonnull Authentication getAuthentication();
 
+    /**
+     * Create {@link Action}s to be attached to {@linkplain #getOwner() owner} when it has UI.
+     */
+    public @Nonnull Collection<? extends Action> createActions() {
```

Comment:
I think you could get rid of this API and `ActionFromFlowExecution` if there were simply a supported way to go from `Run` to `FlowExecution` without linking to `WorkflowRun.getExecutionPromise`, which would be useful elsewhere anyway.